packages/paste-core/components/data-grid/src/DataGridHeader.tsx

```diff
@@ -1,14 +1,18 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import {DataGridCell} from './DataGridCell';
+import {DataGridCell, DataGridCellProps} from './DataGridCell';
 import type {ThProps} from './table/Th';
+
+type DataGridHeaderProps = ThProps & Omit<DataGridCellProps, 'as'>;
```

Comment:
question: Does `thProps` really not expose `colSpan` as a props. It is I believe a global HTML attribute for a `th` element. Maybe we need to fix the types there instead? Potentially `ThProps` might be wrong... 🤔

Reply:
Asked a different way, should the `ThProps` interface in that file extend the types from the native `th` HTML element?

```jsx
Export interface ThProps extends React.HTMLElement<ThElement> {
  ...
}
```

(Note, just random pseudo code I just wrote on my iPad, exact code might differ)

I think the reason this, and future issues with these will crop up, is we're not extended the types of the html element we export a component for. 

The other risk might be that extending the grid cell props as the grid header props, in terms of HTML elements, there might be incorrect overlaps between the two. I.e. a `td` might not share something with a `th` element and vice-a-versa